docs/specifications/tests/DNSSEC-TP/dnssec08.md

```diff
@@ -1,41 +1,191 @@
+It is assumed that *Child Zone* is tested and reported by [Basic04]. This test
+case will just ignore non-responsive name servers or name servers not giving a
+correct DNS response for an authoritative name server.
+
+This test case is only relevant if the zone has been DNSSEC signed.
+
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+
+## Summary
+
+* If no DNSKEY records are found, then further inverstigation will not be done
+  and no messages will be outputted.
+
+Message Tag outputted              | Level   | Arguments          | Description of when message tag is outputted
+:----------------------------------|:--------|:-------------------|:--------------------------------------------
+DS08_ALGO_NOT_SUPPORTED_BY_ZM      | NOTICE  | ns_ip_list, algo_mnemo, algo_num, keytag | This installation of Zonemaster does not support the DNSKEY algorithm.
+DS08_MISSING_RRSIG_IN_RESPONSE     | ERROR   | ns_ip_list         | DNSKEY is unsigned which is against expectation
+DS08_NO_MATCHING_DNSKEY            | ERROR   | ns_ip_list, keytag | DNSKEY RRset is signed with an RRSIG that does not match any DNSKEY
+DS08_RRSIG_NOT_VALID_BY_DNSKEY     | ERROR   | ns_ip_list, keytag | DNSKEY RRset is signed with an RRSIG that cannot be validated by the matching DNSKEY
+DS08_DNSKEY_RRSIG_EXPIRED          | ERROR   | ns_ip_list, keytag | DNSKEY RRset is signed with an RRSIG that has expired
+DS08_DNSKEY_RRSIG_NOT_YET_VALID    | ERROR   | ns_ip_list, keytag | DNSKEY RRsetis signed with a not yet valid RRSIG
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+
+## Test procedure
+
+1.  Create a DNSKEY query with DO flag set for *Child Zone* ("DNSKEY Query").
+
+2.  Retrieve all name server IP addresses for the
+    *Child Zone* using [Method4] and [Method5] ("NS IP").
+
+3.  Create the following empty sets:
+    1.  Name server IP address ("DNSKEY without RRSIG").
+    2.  Name server IP address and RRSIG key tag ("DNSKEY RRSIG not yet valid").
+    3.  Name server IP address and RRSIG key tag ("DNSKEY RRSIG expired").
+    4.  Name server IP address and RRSIG key tag ("No matching DNSKEY").
+    5.  Name server IP address and RRSIG key tag ("RRSIG not valid by DNSKEY").
+    6.  Name server IP address, DNSKEY record key tag and DNSKEY algorithm code
+        ("Algo Not Supported By ZM").
+
+4.  For each name server IP address in *NS IP* do:
+
+    1. Send *DNSKEY Query* to the name server IP.
+    2. If at least one of the following criteria is met, then go to next name
+       server IP:
+         1. There is no DNS response.
+         2. RCODE is not NOERROR.
+         3. AA flag is not set.
+         4. There are no DNSKEY records in the answer section.
```

Comment:
Would you mind align those lines as in other specs?

> 2. The RCODE of response is not "NoError" ([IANA RCODE List]).
> 3. The AA flag is not set in the response.
> 4. There is no DNSKEY record with matching owner name in the answer section.

Reply:
Done.